### DISCUSSION TITLE
Cannot create a package that has no dependencies nor content.

### DISCUSSION THREAD

**Discussion by goreshmusalay** [NuGet/Home#12498] [General]
I am trying to generate NuGet package by using cPack,
I am new to cmake and cpack stuff, so bare with me if the questions and doubt i have are stupid  ones

I am hitting the following error on issuing cpack --build . --package

1. Cannot create a package that has no dependencies nor content.
I have added following NUGET params to the CMakeList.txt file.

1. CPACK_NUGET_PACKAGE_VERSION
2. CPACK_NUGET_PACKAGE_DEPENDENCIES
3. CPACK_NUGET_PACKAGE_DEBUG
4. CPACK_NUGET_PACKAGE_DESCRIPTION
5. CPACK_GENERATOR as NuGet
6. CPACK_NUGET_PACKAGE_NAME

The cpack geneartes the nuget spec file but it has no <files> tag inside it
In build Dir I see folder created with the cmake install (FILES, ....) options that was added.

Can you help in what must i have done wrong here?

**Comment by jeffkl** [comment]
Can you elaborate more on what cpack does and what it is?  We are not familiar with it.

**Reply by goreshmusalay** [reply]
so cpack allows you to make platform agnostic packaging possible, just like cmake, where you can compile code cross platform
https://cmake.org/cmake/help/latest/cpack_gen/nuget.html
in which you can specify which packaging you need.
Above pasted link is the NuGet generator that allows to add the tags to it.

this tags are added in part of CMakeList.txt file as given by the document.
it does go to the steps where it generates the spec file, 
but it fails to pack the `Program| Files` into the package.

If i try to open up the spec file i do see that <files> </files> is missing, i guess its because of that it fails.
I tried to google it around why the <files> are missing, but i didnt found any good explanation around it.

Let me know if we need any more details on the steps, will be happy to share it up.
-G

**Reply by jeffkl** [reply]
Do you know if the problem is in CPack?  It sounds like it is not generating the `.nuspec` you expect.

**Reply by goreshmusalay** [reply]
no sure though, cause cpack mandatory fields were indeed updated in CMakeList.txt.
Let me open up discussion on Cpack side as well just to get some info around it, currently there are no available links that explains this issue online.

**Reply by malaterre** [reply]
Done:

* https://gitlab.kitware.com/cmake/cmake/-/issues/24892
